src/Microsoft.AspNetCore.Mvc.ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -89,6 +80,8 @@ public override void Contextualize(ViewContext viewContext)
                     nameof(viewContext));
             }
 
+            _expressionHelper = viewContext.HttpContext.RequestServices.GetRequiredService<ExpressionHelper>();
```

Comment:
Kinda iffy, but it's the least breaking option. Another alternative is to make the ctor `internal` and use a "factory" method in DI to new this up. I'm not sure it's worth the trouble.

Reply:
Well, if we could break things that much, would be great to make `DefaultHtmlGenerator`, `HtmlHelper` and `HtmlHelper<T>` `internal`. Fall-back would be to make `DefaultHtmlGenerator` and `HtmlHelper<T>` `sealed` and remove any `protected` or `virtual` members not used in test derived classes (if any).